src/util/triplea/MapXMLCreator/CanalDefinitionsPanel.java

```diff
@@ -0,0 +1,499 @@
+package util.triplea.MapXMLCreator;
+
+import games.strategy.util.Tuple;
+
+import java.awt.Color;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.event.ActionEvent;
+import java.awt.event.MouseEvent;
+import java.awt.geom.Rectangle2D;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javax.swing.AbstractAction;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import util.triplea.MapXMLCreator.TerritoryDefinitionDialog.DEFINITION;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class CanalDefinitionsPanel extends ImageScrollPanePanel
+{
+	protected static final String NEW_CANAL_OPTION = "<new Canal>";

```

Comment:
commit first, push that to your branch. Then check if this PR has any merge conflicts: https://github.com/frigoref/triplea/pull/2, if not, you can merge it and it'll update this PR (#485)